tool/tsh/common/tsh.go

```diff
@@ -2615,7 +2615,23 @@ func onLogout(cf *CLIConf) error {
 		fmt.Printf("Logged out %v from %v.\n", cf.Username, proxyHost)
 	// Remove all keys.
 	case proxyHost == "" && cf.Username == "":
-		tc, err := makeClient(cf)
+		proxy, err := cf.getClientStore().CurrentProfile()
+		if err != nil && !trace.IsNotFound(err) {
+			return trace.Wrap(err)
+		}
+
+		// If there's no current profile, try to use the first profile.
+		if proxy == "" && len(profiles) > 0 {
+			proxy = profiles[0].ProxyURL.Host
+		}
+
+		// No current profile and no profiles exist, exit gracefully.
+		if proxy == "" {
+			fmt.Printf("All users logged out.\n")
+			return nil
+		}
```

Comment:
What if we were more explicit about the checks we perform at the cost of increasing the nesting level? If I understand correctly, the logic boils down to this:

```go
		if proxy == "" {
			if len(profiles) > 0 {
				proxy = profiles[0].ProxyURL.Host
			} else {
				fmt.Printf("All users logged out.\n")
				return nil
			}
		}
```

Should we even operate on `cf.getClientStore().CurrentProfile()` instead of using data we already fetched?

```go
		var proxy string
		switch {
		case active != nil:
			proxy = active.ProxyURL.Host
		case len(available) > 0:
			proxy = available[0].ProxyURL.Host
		default:
			fmt.Printf("All users logged out.\n")
			return nil
		}
```